### ISSUE TITLE
Integrate POIs with ESRI

### ISSUE THREAD

**Issue by a6wu** [UCSD/campus-mobile#1092]
ITS-Pro time category: https://its-pro.ucsd.edu/browse/MA-81

This is complete when:

- [ ] The API for querying UCSD POIs are going to ESRI

**Comment by a6wu** [comment]
@gmihir Does this work https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/findAddressCandidates?

**Comment by gmihir** [comment]
Yes, I plan to use the findAddressCandidates operation as a REST endpoint. My main issue is that I don't know how to create the geocode service itself, and I've had trouble finding where to create it on the ESRI web tool.

https://developers.arcgis.com/rest/services-reference/find-address-candidates.htm

**Comment by gmihir** [comment]
Notes on initial attempts:

- The current POIs are located in a hosted feature layer on ESRI online. 
- The main issue surrounds creating a REST endpoint to query POIs based on a given location (e.g. find closest parking lot to user)
- A way to do this is with https://doc.arcgis.com/en/arcgis-online/analyze/find-nearest.htm. This tool fits the desired functionality, but we are having trouble with using our own hosted feature layer and also to use this with a REST endpoint.
- Another option is this operation: https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm. This operation is accessible via REST endpoint and exhibits the functionality we want. However, the issue with this is using our own hosted feature layer. This operation defaults to using ESRI's own POIs, which aren't adequate for our needs. We tried to create a geocode service but still could not integrate this with our hosted feature layer to query the points we wanted to query. 
- This is also a possibility: https://developers.arcgis.com/javascript/latest/api-reference/esri-tasks-Locator.html#suggestLocations, but suffers from the same issues as the other attempts.
